db/tags/utils.hh

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <optional>
+#include <seastar/core/future.hh>
+#include <seastar/core/sstring.hh>
+#include "seastarx.hh"
+
+#include "schema.hh"
+#include "service/client_state.hh"
+#include "service/migration_manager.hh"
+
+namespace db {
+
+// get_tags_of_table() returns all tags associated with the given table, or
+// nothing if the table is missing the tags extension.
+std::optional<std::reference_wrapper<const std::map<sstring, sstring>>> get_tags_of_table(schema_ptr schema);
```

Comment:
Now that this code is out of Alternator it becomes even more important to accurately document this API.  I would ask myself what does the returned reference point to, and how long is it guaranteed to be alive.

By the way, instead of returning an optional<reference>, isn't the more classic approach to use a **pointer**, which is more-or-less an optional reference?

Reply:
We do have quite a few precedents of using a pointer like that in the code, so feel free to use it as well. It's theoretically less safe, but on the other hand it's quite obvious that we need to do a null check on a pointer returned by some helper function, so I consider it "safe enough"